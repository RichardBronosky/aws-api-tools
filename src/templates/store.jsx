import React from 'react'
import Helmet from 'react-helmet'
import { Map, Marker, InfoWindow, GoogleApiWrapper } from 'google-maps-react'
import Layout from '../components/layout'
import Breadcrumb from '../components/shared/breadcrumbs'
import RTGLink from '../components/shared/link'
import { decodeHtml, slugify, weekdays, months, getStandardTime } from '../lib/helpers/string-helper'
import { contentfulImage } from '../lib/helpers/contentful'
import { getStoreHours, getStoreName, getStoreInfoFromFeed } from '../lib/helpers/store-locator'
import { expandState } from '../lib/helpers/geo-location'
import { setupAnalytics } from '../lib/helpers/google-tag-manager'
import StoreInfo from '../components/store-locator/store-info'
import '../assets/css/pages/store-location.sass'

class Store extends React.Component {
  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedMarker: {},
  }

  componentDidMount() {
    const store = this.props.pageContext.store
    setupAnalytics({ pageData: { type: 'store', title: getStoreName(store), path: window.location.pathname } })
  }

  createCrumb = (crumb, slugPath) => {
    return {
      altDesc: `View All ${ crumb } Products`,
      id: slugPath,
      slug: slugPath,
      text: crumb,
    }
  }

  generateStoreCrumbs = store => {
    let crumbs = []
    crumbs.push(this.createCrumb('Find a Showroom', '/stores'))
    if (store.state) {
      crumbs.push(
        this.createCrumb(
          `Find a Showroom - ${ expandState(store.state) }`,
          `/stores/${ slugify(expandState(store.state)) }`
        )
      )
    }
    crumbs.push(
      this.createCrumb(
        decodeHtml(`${ store.city }, ${ store.state }${ store.storeName ? ` (${ store.storeName })` : '' }`),
        window.location.pathname
      )
    )
    return {
      crumbs: crumbs,
    }
  }

  onLocationClick = (props, marker) => {
    const store = this.props.pageContext.store
    const storeInfoFromFeed = getStoreInfoFromFeed(store.storeNumber)
    let fullMarker = {
      ...props,
      title: store.address1,
      name: getStoreName(store),
      lat: store.location.lat,
      lng: store.location.lon,
      city: store.city,
      state: store.state,
      storeNumber: store.storeNumber,
      zip: store.zip,
      phone: storeInfoFromFeed.phone.replace('/', '-'),
      hours: getStoreHours(storeInfoFromFeed.store_hours),
    }
    this.setState({
      selectedMarker: fullMarker,
      activeMarker: marker,
      showingInfoWindow: true,
    })
  }

  onMapClicked = () => {
    if (this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
      })
    }
  }

  getTodayOpen = store => {
    const currentDay = weekdays[new Date().getDay()]
    const month = months[new Date().getMonth()].substr(0, 3)
    const day = new Date().getDate()
    const storeOpen = store.store_hours[`${ currentDay.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ currentDay.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ currentDay }, ${ month } ${ day }: ${
        storeOpen ? `Open from ${ getStandardTime(storeOpen) } to ${ getStandardTime(storeClose) }` : 'Closed'
      }`
    } else {
      ;`${ currentDay }, ${ month } ${ day }`
    }
  }

  getHours = (store, day) => {
    const storeOpen = store.store_hours[`${ day.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ day.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ getStandardTime(storeOpen) } - ${ getStandardTime(storeClose) }`
    }
  }

  render() {
    const { activeMarker, selectedMarker, showingInfoWindow } = this.state
    const store = this.props.pageContext.store
    const address = `${ store.address1 }${ store.address2 ? `, ${ store.address2 }` : '' }, ${ store.city }, ${ store.state } ${
      store.zip
    } `
    const storeInfoFromFeed = getStoreInfoFromFeed(store && store.storeNumber)
    if (store) {
      let storeTypeTitle = `${ store.city }, ${ store.state }`
      if (store.storeType === 'Showroom' || store.storeType === 'Express') {
        storeTypeTitle = storeTypeTitle + ' Furniture & Mattress Store'
      } else if (store.storeType === 'Kids') {
        storeTypeTitle = storeTypeTitle + ' Kids Furniture & Mattress Store'
      } else if (store.storeType === 'Outlet') {
        storeTypeTitle = storeTypeTitle + ' Affordable Furniture Outlet Store'
      } else if (store.storeType === 'Outdoor') {
        storeTypeTitle = storeTypeTitle + ' Patio Furniture: Wicker, Teak Outdoor Furniture Sets'
      }
      return (
        <Layout { ...this.props }>
          <Helmet title={ storeTypeTitle } />
          { store.seo && (
            <>
              { store.seo.pageTitle && <Helmet title={ store.seo.pageTitle } /> }
              { store.seo.canonical && (
                <Helmet
                  link={ [
                    {
                      rel: 'canonical',
                      href: store.seo.canonical,
                    },
                  ] }
                />
              ) }
              { store.seo.metaDescription && (
                <Helmet
                  meta={ [
                    {
                      name: 'description',
                      content: store.seo.metaDescription.metaDescription,
                    },
                  ] }
                />
              ) }
              { store.seo.metaRobots && (
                <Helmet
                  meta={ [
                    {
                      name: 'robots',
                      content: store.seo.metaRobots,
                    },
                  ] }
                />
              ) }
              { store.seo.jsonLdSchema && (
                <Helmet
                  script={ [
                    {
                      type: 'application/ld+json',
                      innerHTML: store.seo.jsonLdSchema.jsonLdSchema,
                    },
                  ] }
                />
              ) }
            </>
          ) }
          <section className="seo-page-heading full-width hide show">
            <div className="grid-container ">
              <div className="grid-x grid-margin-x">
                <div className="cell small-12">
                  <h1>
                    { store.seo && store.seo.pageHeading
                      ? store.seo.pageHeading
                      : `${ storeTypeTitle }${ store.storeName ? ` (${ store.storeName })` : '' }` }
                  </h1>
                </div>
              </div>
            </div>
          </section>
          <div className="stores-breadcrumb">
            <Breadcrumb data={ this.generateStoreCrumbs(store) } includeHeading={ true } />
          </div>
          <div className="store-location">
            <div className="loc-info">
              { address }
              <RTGLink
                data={ {
                  url: `tel:${ storeInfoFromFeed.phone.replace('/', '-') }`,
                  title: 'Call store',
                  category: 'store page',
                  action: 'Call click',
                  label: 'Call',
                } }
                className="phone-link"
              >
                { storeInfoFromFeed.phone.replace('/', '-') }
              </RTGLink>
              <span className="today">{ this.getTodayOpen(storeInfoFromFeed) }</span>
            </div>
            <div className="grid-x map-container">
              <Map
                initialCenter={ { lat: store.location.lat, lng: store.location.lon } }
                center={ { lat: store.location.lat, lng: store.location.lon } }
                className="map"
                google={ this.props.google }
                zoom={ 11 }
                onClick={ this.onMapClicked }
              >
                <Marker
                  position={ { lat: store.location.lat, lng: store.location.lon } }
                  title={ `${ store.city }${ store.storeName ? ` - ${ store.storeName }` : '' } ${ store.storeType }` }
                  icon={ {
                    url: `http://maps.google.com/mapfiles/markerA.png`,
                  } }
                  onClick={ this.onLocationClick }
                />
                <InfoWindow marker={ activeMarker } visible={ showingInfoWindow }>
                  <StoreInfo selectedMarker={ selectedMarker } />
                </InfoWindow>
              </Map>
              { store.storeImage && (
                <div className="store-image-container">
                  <img className="store-image" alt={ storeTypeTitle } src={ contentfulImage(store.storeImage.file.url) } />
                </div>
              ) }
              { storeInfoFromFeed.store_hours && (
                <div className="section-container">
                  <h2 className="section-heading">Store Hours:</h2>
                  { weekdays.map(day => (
                    <div className="hour" key={ day }>
                      { storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">{ this.getHours(storeInfoFromFeed, day) }</span>
                        </>
                      ) }
                      { !storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">Closed</span>
                        </>
                      ) }
                    </div>
                  )) }
                  { store.specialClosings && (
                    <span
                      className="special-closings"
                      dangerouslySetInnerHTML={ { __html: store.specialClosings.markdown.childMarkdownRemark.html } }
                    />
                  ) }
                </div>
              ) }
              { store.directions && (
                <div className="section-container">
                  <h2 className="section-heading">Store Location / Address:</h2>
                  <span>{ address }</span>
                  <br />
                  <span
                    className="directions"
                    dangerouslySetInnerHTML={ { __html: store.directions.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.neighborhoodsNearby && store.neighborhoodsNearby.length > 0 && (
                <div className="section-container">
                  <h2 className="section-heading">Neighborhoods Nearby:</h2>
                  <ul>
                    { store.neighborhoodsNearby.map((hood, index) => (
                      <li key={ index } className="hood">
                        { hood }
                      </li>
                    )) }
                  </ul>
                </div>
              ) }
              { store.description && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">Store Description:</h2>
                  <span
                    className="description"
                    dangerouslySetInnerHTML={ { __html: store.description.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.faqs && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">FAQs:</h2>
                  <span
                    className="faqs"
                    dangerouslySetInnerHTML={ { __html: store.faqs.markdown.childMarkdownRemark.html } }
                  />
                </div>
              ) }
            </div>
          </div>
        </Layout>
      )
    } else {
      return null
    }
  }
}

export default GoogleApiWrapper({
  apiKey: `${ process.env.GATSBY_GOOGLE_BROWSER_API_KEY }`,
})(Store)
