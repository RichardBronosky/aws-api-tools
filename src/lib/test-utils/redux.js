import configureMockStore from 'redux-mock-store'
import * as reduxStore from '../../redux/store'

const mockStore = configureMockStore([])

export const mockStoreDispatch = initialState => {
  reduxStore.store = mockStore(initialState)
  reduxStore.store.dispatch = jest.fn()
}
