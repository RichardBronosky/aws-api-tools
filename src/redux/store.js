import React from 'react'
import { Provider } from 'react-redux'
import { createStore, compose, combineReducers } from 'redux'

import global from './modules/global'
import location from './modules/location'
import cart from './modules/cart'
import checkout from './modules/checkout'

const rootReducer = combineReducers({
  global,
  location,
  cart,
  checkout,
})

// Redux DevTools
const windowObject = typeof window !== 'undefined' && window
const reduxDevTools =
  process.env.NODE_ENV === 'development' && windowObject.devToolsExtension
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : _ => _

const configureStore = initialState => createStore(rootReducer, initialState, compose(reduxDevTools))

export const store = configureStore()

// eslint-disable-next-line react/display-name,react/prop-types
export default ({ element }) => <Provider store={ store }>{ element }</Provider>
