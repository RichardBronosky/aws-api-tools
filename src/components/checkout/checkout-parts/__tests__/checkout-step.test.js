import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '../../../../redux/store'
import CheckoutStep from '../checkout-step'
import ShippingSection from '../shipping-section'
import DeliverySection from '../delivery-section'
import PaymentSection from '../payment-section'
import ReviewSection from '../review-section'

describe('CheckoutStep', () => {
  const shippingProps = {
    sectionTitle: 'Shipping Address',
    sectionType: 'shipping',
    sectionNumber: '1',
    nextSection: 'delivery',
  }

  it('renders shipping step correctly', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CheckoutStep { ...shippingProps }>
            <ShippingSection />
          </CheckoutStep>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  const deliveryProps = {
    sectionTitle: 'Delivery',
    sectionType: 'delivery',
    sectionNumber: '2',
    previousSection: 'shipping',
    nextSection: 'payment',
  }

  it('renders delivery step correctly', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CheckoutStep { ...deliveryProps }>
            <DeliverySection />
          </CheckoutStep>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  const paymentProps = {
    sectionTitle: 'Payment',
    sectionType: 'payment',
    sectionNumber: '3',
    previousSection: 'delivery',
    nextSection: 'review',
  }

  it('renders payment step correctly', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CheckoutStep { ...paymentProps }>
            <PaymentSection />
          </CheckoutStep>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  const reviewProps = {
    sectionTitle: 'Review Order',
    sectionType: 'review',
    sectionNumber: '4',
    previousSection: 'payment',
  }

  it('renders review step correctly', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CheckoutStep { ...reviewProps }>
            <ReviewSection />
          </CheckoutStep>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
