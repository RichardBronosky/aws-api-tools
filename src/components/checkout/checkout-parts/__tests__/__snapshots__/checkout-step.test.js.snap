// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutStep renders delivery step correctly 1`] = `
<div
  className="section"
>
  <div
    className="section-header"
  >
    <i
      className="icon circle"
    >
      <p
        className="section-number"
      >
        2
      </p>
    </i>
    <h2
      className="section-title"
    >
      Delivery
    </h2>
    <button
      aria-label="Complete shipping step before proceeding to Delivery"
      className="edit-btn disabled"
      disabled={true}
      onClick={[Function]}
      value="Edit"
    >
      EDIT 
      <span
        className="hide508"
      >
        Delivery
      </span>
    </button>
  </div>
  <div
    className="section-content"
  />
</div>
`;

exports[`CheckoutStep renders payment step correctly 1`] = `
<div
  className="section"
>
  <div
    className="section-header"
  >
    <i
      className="icon circle"
    >
      <p
        className="section-number"
      >
        3
      </p>
    </i>
    <h2
      className="section-title"
    >
      Payment
    </h2>
    <button
      aria-label="Complete delivery step before proceeding to Payment"
      className="edit-btn disabled"
      disabled={true}
      onClick={[Function]}
      value="Edit"
    >
      EDIT 
      <span
        className="hide508"
      >
        Payment
      </span>
    </button>
  </div>
  <div
    className="section-content"
  />
</div>
`;

exports[`CheckoutStep renders review step correctly 1`] = `
<div
  className="section"
>
  <div
    className="section-header"
  >
    <i
      className="icon circle"
    >
      <p
        className="section-number"
      >
        4
      </p>
    </i>
    <h2
      className="section-title"
    >
      Review Order
    </h2>
    <button
      aria-label="Complete payment step before proceeding to Review Order"
      className="edit-btn disabled"
      disabled={true}
      onClick={[Function]}
      value="Edit"
    >
      EDIT 
      <span
        className="hide508"
      >
        Review Order
      </span>
    </button>
  </div>
  <div
    className="section-content"
  />
</div>
`;

exports[`CheckoutStep renders shipping step correctly 1`] = `
<div
  className="section"
>
  <div
    className="section-header"
  >
    <i
      className="icon circle"
    >
      <p
        className="section-number"
      >
        1
      </p>
    </i>
    <h2
      className="section-title"
    >
      Shipping Address
    </h2>
  </div>
  <div
    className="section-content"
  >
    <form
      className="shipping-form"
    >
      <p
        className="required-label"
      >
        *Asterisks indicate required fields in the form below
      </p>
      <label
        className="label firstName"
        htmlFor="firstName"
      >
        First Name*
         
        <input
          aria-invalid={false}
          aria-required={true}
          checked={false}
          id="firstName"
          name="First Name"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="First Name"
          type="text"
          value=""
        />
        <span />
      </label>
      <label
        className="label lastName"
        htmlFor="lastName"
      >
        Last Name*
         
        <input
          aria-invalid={false}
          aria-required={true}
          checked={false}
          id="lastName"
          name="Last Name"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Last Name"
          type="text"
          value=""
        />
        <span />
      </label>
      <span
        className="hide508"
        id="addressInstructions"
      >
        Start typing to receive address suggestions.
      </span>
      <span
        className="hide508"
        id="addressInstructions"
      >
        Start typing to receive address suggestions.
      </span>
      <label
        htmlFor="address"
      >
        <span
          className="address-label"
        >
          Address
          <span
            className="hide508"
          >
             Lookup
          </span>
          *
        </span>
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-controls="addressSuggest"
            aria-describedby="addressInstructions"
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-label="Address Lookup"
            aria-required={true}
            autoComplete="new-password"
            className="address-lookup-input"
            name="address-lookup"
            onBlur={[Function]}
            onBlurCapture={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Address Lookup*"
            role="combobox"
            type="text"
            value=""
          />
        </div>
        <div
          className="manual-address"
        >
          <a
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            Enter address manually.
          </a>
        </div>
      </label>
      <label
        className="label phone"
        htmlFor="phone"
      >
        Phone Number*
         
        <span
          className="hide508"
        >
          Numbers and hyphens only
        </span>
        <input
          aria-invalid={false}
          aria-required={true}
          checked={false}
          id="phone"
          name="Phone Number"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Phone Number"
          type="tel"
          value=""
        />
        <span />
      </label>
      <label
        className="label altPhone"
        htmlFor="altPhone"
      >
        Alternate Phone
         
        <input
          aria-invalid={null}
          checked={false}
          id="altPhone"
          name="Alternate Phone"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Alternate Phone"
          type="tel"
          value=""
        />
        <span />
      </label>
      <label
        className="label email"
        htmlFor="email"
      >
        Email*
         
        <input
          aria-invalid={false}
          aria-required={true}
          checked={false}
          id="email"
          name="Email"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <span />
      </label>
      <fieldset
        className="text-me-container"
      >
        <legend
          className="text-me-label"
        >
          Text me delivery time frames
        </legend>
        <input
          aria-invalid={null}
          checked={true}
          className="text-me"
          id="deliveryTexts"
          name="text-me"
          onChange={[Function]}
          onClick={[Function]}
          placeholder=""
          type="radio"
          value={true}
        />
        <span>
          <span
            className="radio-label"
          >
            Yes
          </span>
        </span>
        <input
          aria-invalid={null}
          checked={false}
          className="text-me"
          id="deliveryTexts"
          name="text-me"
          onChange={[Function]}
          onClick={[Function]}
          placeholder=""
          type="radio"
          value={false}
        />
        <span>
          <span
            className="radio-label"
          >
            No
          </span>
        </span>
      </fieldset>
      <br />
      <div
        className="checkbox-container"
      >
        <input
          aria-invalid={null}
          checked={true}
          className="text-me"
          id="emailCampaign"
          name="Sign up for Rooms To Go email promotions and news letters"
          onChange={[Function]}
          onClick={[Function]}
          placeholder=""
          type="checkbox"
          value=""
        />
        <span>
          <p>
            Sign up for Rooms To Go email promotions and news letters
          </p>
        </span>
      </div>
    </form>
    <div
      className="btn-container"
    >
      <button
        className="continue-btn"
        onClick={[Function]}
        value="Continue"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
